add(docs): how to make QDT work with private git repositories (#634)
qgis-deployment/qgis-deployment-toolbelt-cli: docs/guides/howto_manage_private_git.md

# How to make QDT work with a private Git repository

> For now, QDT does not support private Git repositories. It's a "to be funded" issue. If you are willing to see this feature land in the functional perimeter, don't hesitet to get in touch!

Reviewer: ```suggestion
For now, QDT does not support private Git repositories. It's a "to be funded" issue. If you'd like to see this feature included in our functional scope, please don't hesitate to get in touch!
```

```{mermaid}
---
title: QDT workflow around a Git private forge (to be funded)
---

flowchart LR
    A["QGIS/QDT admin"] <---> |"classic git workflow<br/>clone/pull & git push"| D(("Private Git repository<br/>i.e. git.myorg.com or on GitHub/Lab"))
    D --->|"QDT"| F["QGIS end-user<br/>profiles"]
```

> In the meanwhile, it's not a blocking lack, since QDT support local git repositories, i.e. stored on a network drive. Here comes a diagram illustrating the typical workflow used by most of end-users:

Reviewer: ```suggestion
In the meantime, this isn't a critical limitation, as QDT supports local git repositories stored on network drives. Here comes a diagram illustrating the typical workflow used by most of end-users:
```

```{mermaid}
---
title: QDT workflow around a Git private forge (workaround)
---

flowchart LR
    A["QGIS/QDT admin"] <---> |"classic git workflow<br/>clone/pull & git push"| D(("Private Git repository<br/>i.e. git.myorg.com or on GitHub/Lab"))
    D -->|"Pull<br/>(with deployment token)"| E[["Server on local network<br/>i.e. '//gis/software/qgis/qdt/profiles/'"]]
    E --->|"QDT"| F["QGIS end-user<br/>profiles"]
```

> To perform the intermediary git pull, it depends on the organization and GIS team habits. It can be performed manually through the command-line, with a GUI like the excellent [GitHub Desktop](https://github.com/apps/desktop) or with a script. We give below an example in PowerShell for Windows.

Reviewer: ```suggestion
To execute the intermediate git pull, the approach depends on the organization and GIS team habits. This can be done manually through the command-line, with a GUI like the excellent [GitHub Desktop](https://github.com/apps/desktop) or with a script. Below is an example PowerShell script for Windows.
```

:::{info}

> This script is a sample and might not comply with your environment and/or IT policy. If you intend to use it in production, take time to review it before. If you improve or fix it, please share it.

Reviewer: ```suggestion
This script is provided as a sample and may not fully comply with your environment or IT policies. Before implementing in production, take time to review and test it in your environment. If you make improvements or fixes, please share them.
```
